crates/wash-lib/src/cli/link.rs

```diff
@@ -4,19 +4,19 @@ use wasmcloud_control_interface::LinkDefinition;
 
 use crate::{
     cli::{labels_vec_to_hashmap, CliConnectionOpts},
-    common::boxed_err_to_anyhow,
+    common::{boxed_err_to_anyhow, find_actor_id, find_provider_id},
     config::WashConnectionOptions,
-    id::{ModuleId, ServiceId},
 };
 
 #[derive(Parser, Debug, Clone)]
 pub struct LinkDelCommand {
     #[clap(flatten)]
     pub opts: CliConnectionOpts,
 
-    /// Public key ID of actor
-    #[clap(name = "actor-id", value_parser)]
-    pub actor_id: ModuleId,
+    /// Public key ID or name of actor to match on. If an actor name is given and matches multiple
+    /// actors, an error will be returned with a list of matching actors and their IDs.
+    #[clap(name = "actor-id")]
+    pub actor_id: String,
```

Comment:
NIT: Feels like we lost some specificity/protection here... I would have liked an enum to capture both these cases

Reply:
I actually don't agree here. All of these go through processing after they are specified that processes them to `ModuleId` (or their relevant type). We need this loose so we can take all sorts of match-y things and having it parse to an enum feels like extra boilerplate with not a lot of benefit

Reply:
Ah it feels here like when we try to use these values elsewhere we'll want to know if it is one thing or the other (a "fuzzy" name or an explicit ID, or something else). Feels like the enum can also hold information as we enrich it/figure out what the thing is (if something is a provider fuzzy name it's probably going to stay the same).

All that said I'm OK with seeing how this evolves over time -- biasing for speed.